packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -644,6 +670,36 @@ abstract class ClusterBase extends Resource implements ICluster {
   public addCdk8sChart(id: string, chart: cdk8s.Chart): KubernetesManifest {
     return this.addManifest(id, ...chart.toJson());
   }
+
+  public addServiceAccount(id: string, options: ServiceAccountOptions = {}): ServiceAccount {
+    return new ServiceAccount(this, id, {
+      ...options,
+      cluster: this,
+    });
+  }
+
+  /**
+   * An `OpenIdConnectProvider` resource associated with this cluster, and which can be used
+   * to link this cluster to AWS IAM.
+   *
+   * A provider will only be defined if this property is accessed (lazy initialization).
+   */
+  public get openIdConnectProvider() {
```

Comment:
We should not be implicitly creating providers for imported clusters (we said that the user will always provide an instance of `IOpenConenctIdProvider` to the imported cluster). So this should be moved back down to `Cluster`.

Reply:
sure yeah